python/hopsworks_common/execution.py

```diff
@@ -0,0 +1,258 @@
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+import json
+
+import humps
+from hopsworks_common import client, constants, util
+from hopsworks_common.client.exceptions import JobExecutionException
+from hopsworks_common.core import execution_api
+from hopsworks_common.engine import execution_engine
+
+
+class Execution:
+    def __init__(
+        self,
+        id=None,
+        state=None,
+        final_status=None,
+        submission_time=None,
+        stdout_path=None,
+        stderr_path=None,
+        app_id=None,
+        hdfs_user=None,
+        args=None,
+        progress=None,
+        user=None,
+        files_to_remove=None,
+        duration=None,
+        flink_master_url=None,
+        monitoring=None,
+        type=None,
+        href=None,
+        job=None,
+        **kwargs,
+    ):
+        self._id = id
+        self._final_status = final_status
+        self._state = state
+        self._submission_time = submission_time
+        self._stdout_path = stdout_path
+        self._stderr_path = stderr_path
+        self._args = args
+        self._progress = progress
+        self._user = user
+        self._duration = duration
+        self._monitoring = monitoring
+        self._app_id = app_id
+        self._hdfs_user = hdfs_user
+        self._job = job
+
+        self._execution_engine = execution_engine.ExecutionEngine()
+        self._execution_api = execution_api.ExecutionApi()
+
+    @classmethod
+    def from_response_json(cls, json_dict, job):
+        json_decamelized = humps.decamelize(json_dict)
+        if "count" not in json_decamelized:
+            return cls(**json_decamelized, job=job)
+        elif json_decamelized["count"] == 0:
+            return []
+        else:
+            return [
+                cls(**execution, job=job) for execution in json_decamelized["items"]
+            ]
+
+    def update_from_response_json(self, json_dict):
+        json_decamelized = humps.decamelize(json_dict)
+        self.__init__(**json_decamelized)
+        return self
+
+    @property
+    def id(self):
+        """Id of the execution"""
+        return self._id
+
+    @property
+    def job_name(self):
+        """Name of the job the execution belongs to"""
+        return self._job.name
+
+    @property
+    def job_type(self):
+        """Type of the job the execution belongs to"""
+        return self._job.job_type
+
+    @property
+    def state(self):
```

Comment:
Do you know what exactly it can be? There are `contants.JOBS.SUCCESS_STATES` and `constants.JOBS.ERROR_STATES`, but they don't include in-progress states.

Reply:
Ok, I have found this: https://github.com/logicalclocks/hopsworks/blob/29d2bfe8a122821c6df2a5755601326536af31b1/hopsworks-persistence/src/main/java/io/hops/hopsworks/persistence/entity/jobs/configuration/history/JobState.java#L49